### PR TITLE
Add wireframe-verification crate with placeholder Stateright model (15.1.2)

### PR DESCRIPTION
**Pull request by leynos**
## Summary
Introduce an internal crate for formal verification support: `crates/wireframe-verification`. It provides a shared Stateright harness, a placeholder connection model, and tests to exercise the verification workflow. This lays the groundwork for roadmap item 15.1.2 and later extensions toward the real ConnectionActor model.

## Changes
- New crate: `crates/wireframe-verification`
  - lib.rs exporting // modules for the verification model and harness
  - connection_model:
    - action.rs: placeholder ConnectionAction enum
    - mod.rs / model.rs / state.rs / properties.rs: placeholder semantic model and helpers
  - harness.rs: shared bounded Stateright checker builder and assertion helpers
  - tests:
    - connection_actor.rs: integration test for the placeholder model
    - verification_harness.rs: integration test for the shared harness with explicit bounds
- Workspace and manifest updates
  - Root `Cargo.toml`: add `crates/wireframe-verification` to workspace members; keep default member as root
  - Documentation alignment to reflect new internal crate and its role in the verification roadmap
  - Tests and fixtures updated to reflect the workspace manifest changes (workspace_members now include the verification crate)
- Documentation artifacts
  - Added ExecPlan doc at `docs/execplans/15-1-2-wireframe-verification-crate.md` describing purpose, constraints, and validation goals
  - Updated docs: `docs/formal-verification-methods-in-wireframe.md`, `docs/roadmap.md`, and related sections to reflect the new cradle and its scope

## Why
This implements the 15.1.2 milestone: an internal, non-published crate to host Stateright models and a reusable bounded-checker harness, enabling safe, repeatable verification workflows while the real ConnectionActor model matures in follow-up roadmap items.

## How to test
- Run crate-specific tests:
  - cargo test -p wireframe-verification
- Run repository tests (workspace):
  - cargo test --workspace
- Verify workspace metadata reflects the new member:
  - cargo metadata --no-deps --format-version 1

## Next steps
- Extend the placeholder connection model toward the real ConnectionActor semantics.
- Expand harness coverage and add more rigorous tests as roadmap items progress.
- Keep documentation in-sync with evolving verification capabilities and workspace layout.

## Validation plan (from ExecPlan)
- [x] Ensure `cargo test -p wireframe-verification` passes
- [x] Ensure workspace manifest tests reflect new member and defaults
- [x] Keep crate non-published (`publish = false`)
- [x] Update developer docs to describe the new verification crate structure

◳ Generated by [DevBoxer](https://www.devboxer.com) ◰

---

ℹ️ Tag @devboxerhub to ask questions and address PR feedback

📎 **Task**: https://www.devboxer.com/task/7849551b-1b21-4381-a196-34d9041706d2

## Summary by Sourcery

Introduce an internal `wireframe-verification` crate for Stateright-based formal models and a shared verification harness, and wire it into the workspace, tests, and documentation as the home for connection-actor verification work.

New Features:
- Add the internal `wireframe-verification` crate providing a placeholder connection-actor Stateright model and shared bounded-checker harness.

Enhancements:
- Extend workspace manifest tests and BDD scenarios to assert the verification crate is a workspace member while keeping the root package as the sole default member.
- Refine workspace manifest support utilities to fetch generic Cargo package IDs for both the root and verification crates.

Build:
- Update the root `Cargo.toml` workspace members to include `crates/wireframe-verification` without changing default members.

Documentation:
- Add an ExecPlan documenting roadmap item 15.1.2 and update formal verification, developer guide, and roadmap docs to describe the new verification crate, workspace layout, and Stateright version.
- Clarify documentation around `wireframe` dependency features and workspace metadata behavior after adding the verification crate.

Tests:
- Add integration tests for the placeholder connection model and shared verification harness in the new crate using `rstest`.
- Update workspace manifest regression tests and BDD feature/step definitions to validate the presence of the verification crate in Cargo metadata and workspace membership.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/wireframe-verification/src/connection_model/model.rs](https://github.com/leynos/wireframe/pull/527/files/61c6e3e2044aab2eec16703be84ff0f177e1b659#diff-10fe5639a0c6fa652948f9549eddd1cfea953578f6da41a75df31e86554681a2)

Comment on lines +42 to +122

~~~rust
    fn next_state(&self, state: &Self::State, action: Self::Action) -> Option<Self::State> {
        let mut next = state.clone();
        next.steps = next.steps.saturating_add(1);

        match action {
            ConnectionAction::EnqueueHigh if !state.high_priority_queued => {
                next.high_priority_queued = true;
                Some(next)
            }
            ConnectionAction::EnqueueLow if !state.low_priority_queued => {
                next.low_priority_queued = true;
                Some(next)
            }
            ConnectionAction::InstallResponse
                if !state.shutdown_requested
                    && matches!(state.active_output, ActiveOutput::Idle) =>
            {
                next.active_output = ActiveOutput::Response;
                Some(next)
            }
            ConnectionAction::InstallMultiPacket
                if !state.shutdown_requested
                    && matches!(state.active_output, ActiveOutput::Idle) =>
            {
                next.active_output = ActiveOutput::MultiPacket;
                next.multi_packet_terminal_count = 0;
                Some(next)
            }
            ConnectionAction::EmitQueued if state.high_priority_queued => {
                next.high_priority_queued = false;
                next.emitted_high_priority = true;
                next.fairness_allows_low = true;
                Some(next)
            }
            ConnectionAction::EmitQueued
                if state.low_priority_queued
                    && (!state.high_priority_queued || state.fairness_allows_low) =>
            {
                next.low_priority_queued = false;
                next.emitted_low_priority = true;
                next.fairness_allows_low = false;
                Some(next)
            }
            ConnectionAction::EmitActiveFrame
                if !matches!(state.active_output, ActiveOutput::Idle) =>
            {
                if state.shutdown_requested {
                    next.shutdown_during_output = true;
                }
                Some(next)
            }
            ConnectionAction::CompleteActiveOutput
                if matches!(state.active_output, ActiveOutput::Response) =>
            {
                next.active_output = ActiveOutput::Idle;
                next.response_completed = true;
                Some(next)
            }
            ConnectionAction::CompleteActiveOutput
                if matches!(state.active_output, ActiveOutput::MultiPacket) =>
            {
                next.active_output = ActiveOutput::Idle;
                next.multi_packet_completed = true;
                next.multi_packet_terminal_count =
                    next.multi_packet_terminal_count.saturating_add(1);
                Some(next)
            }
            ConnectionAction::Shutdown if !state.shutdown_requested => {
                next.shutdown_requested = true;
                if !matches!(state.active_output, ActiveOutput::Idle) {
                    next.shutdown_during_output = true;
                }
                Some(next)
            }
            ConnectionAction::TickFairness if state.low_priority_queued => {
                next.fairness_allows_low = true;
                Some(next)
            }
            _ => None,
        }
    }
~~~

❌ New issue: [Complex Method](https://codescene.io/projects/68308/delta?repo-id=1125256&review-id=527&biomarker=Complex+Method&filename=crates%2Fwireframe-verification%2Fsrc%2Fconnection_model%2Fmodel.rs&method=Model.next_state)
Model.next_state has a cyclomatic complexity of 19, threshold = 9

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/wireframe-verification/src/connection_model/model.rs](https://github.com/leynos/wireframe/pull/527/files/61c6e3e2044aab2eec16703be84ff0f177e1b659#diff-10fe5639a0c6fa652948f9549eddd1cfea953578f6da41a75df31e86554681a2)

Comment on lines +77 to +78

~~~rust
                if state.low_priority_queued
                    && (!state.high_priority_queued || state.fairness_allows_low) =>
~~~

❌ New issue: [Complex Conditional](https://codescene.io/projects/68308/delta?repo-id=1125256&review-id=527&biomarker=Complex+Conditional&filename=crates%2Fwireframe-verification%2Fsrc%2Fconnection_model%2Fmodel.rs&method=Model.next_state)
Model.next_state has 1 complex conditionals with 2 branches, threshold = 2

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[crates/wireframe-verification/src/connection_model/model.rs](https://github.com/leynos/wireframe/pull/527/files/61c6e3e2044aab2eec16703be84ff0f177e1b659#diff-10fe5639a0c6fa652948f9549eddd1cfea953578f6da41a75df31e86554681a2)

Comment on lines +42 to +122

~~~rust
    fn next_state(&self, state: &Self::State, action: Self::Action) -> Option<Self::State> {
        let mut next = state.clone();
        next.steps = next.steps.saturating_add(1);

        match action {
            ConnectionAction::EnqueueHigh if !state.high_priority_queued => {
                next.high_priority_queued = true;
                Some(next)
            }
            ConnectionAction::EnqueueLow if !state.low_priority_queued => {
                next.low_priority_queued = true;
                Some(next)
            }
            ConnectionAction::InstallResponse
                if !state.shutdown_requested
                    && matches!(state.active_output, ActiveOutput::Idle) =>
            {
                next.active_output = ActiveOutput::Response;
                Some(next)
            }
            ConnectionAction::InstallMultiPacket
                if !state.shutdown_requested
                    && matches!(state.active_output, ActiveOutput::Idle) =>
            {
                next.active_output = ActiveOutput::MultiPacket;
                next.multi_packet_terminal_count = 0;
                Some(next)
            }
            ConnectionAction::EmitQueued if state.high_priority_queued => {
                next.high_priority_queued = false;
                next.emitted_high_priority = true;
                next.fairness_allows_low = true;
                Some(next)
            }
            ConnectionAction::EmitQueued
                if state.low_priority_queued
                    && (!state.high_priority_queued || state.fairness_allows_low) =>
            {
                next.low_priority_queued = false;
                next.emitted_low_priority = true;
                next.fairness_allows_low = false;
                Some(next)
            }
            ConnectionAction::EmitActiveFrame
                if !matches!(state.active_output, ActiveOutput::Idle) =>
            {
                if state.shutdown_requested {
                    next.shutdown_during_output = true;
                }
                Some(next)
            }
            ConnectionAction::CompleteActiveOutput
                if matches!(state.active_output, ActiveOutput::Response) =>
            {
                next.active_output = ActiveOutput::Idle;
                next.response_completed = true;
                Some(next)
            }
            ConnectionAction::CompleteActiveOutput
                if matches!(state.active_output, ActiveOutput::MultiPacket) =>
            {
                next.active_output = ActiveOutput::Idle;
                next.multi_packet_completed = true;
                next.multi_packet_terminal_count =
                    next.multi_packet_terminal_count.saturating_add(1);
                Some(next)
            }
            ConnectionAction::Shutdown if !state.shutdown_requested => {
                next.shutdown_requested = true;
                if !matches!(state.active_output, ActiveOutput::Idle) {
                    next.shutdown_during_output = true;
                }
                Some(next)
            }
            ConnectionAction::TickFairness if state.low_priority_queued => {
                next.fairness_allows_low = true;
                Some(next)
            }
            _ => None,
        }
    }
~~~

❌ New issue: [Bumpy Road Ahead](https://codescene.io/projects/68308/delta?repo-id=1125256&review-id=527&biomarker=Bumpy+Road+Ahead&filename=crates%2Fwireframe-verification%2Fsrc%2Fconnection_model%2Fmodel.rs&method=Model.next_state)
Model.next_state has 3 blocks with nested conditional logic. Any nesting of 2 or deeper is considered. Threshold is 2 blocks per function

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response. Ensure that this is validated against the *current* version of the codegraph.

If further refinement to address this finding would be deleterious, please supply a clear explanatory one to two paragraph markdown message I can paste into the CodeScene web ui's diagnostic suppression function so this diagnostic can be silenced.

[crates/wireframe-verification/src/connection_model/model.rs](https://github.com/leynos/wireframe/pull/527/files/4e956971b6d8d7030947911d0170f6801888fccf#diff-10fe5639a0c6fa652948f9549eddd1cfea953578f6da41a75df31e86554681a2)

Comment on lines +64 to +73

~~~rust
    fn apply_emit_high(state: &ConnectionState) -> Option<ConnectionState> {
        if !state.high_priority_queued {
            return None;
        }
        let mut next = Self::stepped(state);
        next.high_priority_queued = false;
        next.emitted_high_priority = true;
        next.fairness_allows_low = true;
        Some(next)
    }
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/68308/delta?repo-id=1125256&review-id=527&biomarker=Code+Duplication&filename=crates%2Fwireframe-verification%2Fsrc%2Fconnection_model%2Fmodel.rs&method=)
The module contains 4 functions with similar structure: PlaceholderConnectionModel.apply_emit_active_frame,PlaceholderConnectionModel.apply_emit_high,PlaceholderConnectionModel.apply_emit_low,PlaceholderConnectionModel.apply_shutdown

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for the following concern and provide an AI coding agent prompt for the fix:

~~~
error: Avoid calling expect on `std::result::Result<std::sync::MutexGuard<'_, tokio::time::Instant>, std::sync::PoisonError<std::sync::MutexGuard<'_, tokio::time::Instant>>>` outside test-only code.
Error:    --> src/fairness.rs:174:27
    |
174 |             let mut now = self.now.lock().expect("MockClock mutex poisoned");
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: The call originates within function `advance` which is not recognised as a test.
    = help: Handle the `Err` variant of `std::result::Result<std::sync::MutexGuard<'_, tokio::time::Instant>, std::sync::PoisonError<std::sync::MutexGuard<'_, tokio::time::Instant>>>` or move the code into a test.
    = note: `#[deny(no_expect_outside_tests)]` on by default

error: Avoid calling expect on `std::result::Result<std::sync::MutexGuard<'_, tokio::time::Instant>, std::sync::PoisonError<std::sync::MutexGuard<'_, tokio::time::Instant>>>` outside test-only code.
Error:    --> src/fairness.rs:181:37
    |
181 |         fn now(&self) -> Instant { *self.now.lock().expect("MockClock mutex poisoned") }
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: The call originates within function `now` which is not recognised as a test.
    = help: Handle the `Err` variant of `std::result::Result<std::sync::MutexGuard<'_, tokio::time::Instant>, std::sync::PoisonError<std::sync::MutexGuard<'_, tokio::time::Instant>>>` or move the code into a test.

error: Avoid calling expect on `std::result::Result<std::vec::Vec<u8>, bincode::error::EncodeError>` outside test-only code.
Error:    --> src/fragment/payload.rs:190:23
    |
190 |         let encoded = encode_to_vec(header, config::standard()).expect("encode header");
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: The call originates within function `assert_fragment_decode_error` which is not recognised as a test.
    = help: Handle the `Err` variant of `std::result::Result<std::vec::Vec<u8>, bincode::error::EncodeError>` or move the code into a test.

error: Avoid calling expect on `std::result::Result<message_assembler::header::ParsedFrameHeader, std::io::Error>` outside test-only code.
Error:    --> src/message_assembler/tests.rs:186:5
    |
186 | /     TestAssembler
187 | |         .parse_frame_header(payload)
188 | |         .expect("header parse")
    | |_______________________________^
    |
    = note: The call originates within function `parse_header` which is not recognised as a test.
    = help: Handle the `Err` variant of `std::result::Result<message_assembler::header::ParsedFrameHeader, std::io::Error>` or move the code into a test.

error: Avoid calling expect on `std::option::Option<std::num::NonZero<usize>>` outside test-only code.
Error:   --> src/message_assembler/state_tests.rs:28:9
   |
28 |         NonZeroUsize::new(1024).expect("non-zero"),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: The call originates within function `state_with_defaults` which is not recognised as a test.
   = help: Handle the `None` variant of `std::option::Option<std::num::NonZero<usize>>` or move the code into a test.

error: Avoid calling expect on `std::option::Option<std::num::NonZero<usize>>` outside test-only code.
Error:   --> src/message_assembler/budget_tests.rs:27:37
   |
27 | fn nz(val: usize) -> NonZeroUsize { NonZeroUsize::new(val).expect("non-zero") }
   |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: The call originates within function `nz` which is not recognised as a test.
   = help: Handle the `None` variant of `std::option::Option<std::num::NonZero<usize>>` or move the code into a test.

error: Avoid calling expect on `std::result::Result<message_assembler::types::FirstFrameInput<'_>, message_assembler::types::FirstFrameInputError>` outside test-only code.
Error:   --> src/message_assembler/budget_tests.rs:40:13
   |
40 |             FirstFrameInput::new(&$hdr, EnvelopeRouting::default(), vec![], $body).expect("valid");
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
51 |     create_first_frame_input!(_header, input, key, body, is_last);
   |     ------------------------------------------------------------- in this macro invocation
   |
   = note: The call originates within function `submit_first` which is not recognised as a test.
   = help: Handle the `Err` variant of `std::result::Result<message_assembler::types::FirstFrameInput<'_>, message_assembler::types::FirstFrameInputError>` or move the code into a test.
   = note: this error originates in the macro `create_first_frame_input` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Avoid calling expect on `std::result::Result<message_assembler::types::FirstFrameInput<'_>, message_assembler::types::FirstFrameInputError>` outside test-only code.
Error:   --> src/message_assembler/budget_tests.rs:40:13
   |
40 |             FirstFrameInput::new(&$hdr, EnvelopeRouting::default(), vec![], $body).expect("valid");
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
62 |     create_first_frame_input!(_header, input, key, body, false);
   |     ----------------------------------------------------------- in this macro invocation
   |
   = note: The call originates within function `submit_first_at` which is not recognised as a test.
   = help: Handle the `Err` variant of `std::result::Result<message_assembler::types::FirstFrameInput<'_>, message_assembler::types::FirstFrameInputError>` or move the code into a test.
   = note: this error originates in the macro `create_first_frame_input` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Avoid calling expect on `std::result::Result<std::sync::MutexGuard<'_, std::vec::Vec<tokio::time::Instant>>, std::sync::PoisonError<std::sync::MutexGuard<'_, std::vec::Vec<tokio::time::Instant>>>>` outside test-only code.
Error:    --> src/server/runtime/tests.rs:142:17
    |
142 |                 call_log.lock().expect("lock").push(Instant::now());
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: The call originates within function `setup_backoff_mock_listener` which is not recognised as a test.
    = help: Handle the `Err` variant of `std::result::Result<std::sync::MutexGuard<'_, std::vec::Vec<tokio::time::Instant>>, std::sync::PoisonError<std::sync::MutexGuard<'_, std::vec::Vec<tokio::time::Instant>>>>` or move the code into a test.

error: Avoid calling expect on `std::result::Result<std::net::SocketAddr, std::net::AddrParseError>` outside test-only code.
Error:    --> src/server/runtime/tests.rs:149:26
    |
149 |         .returning(|| Ok("127.0.0.1:0".parse().expect("addr parse")))
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: The call originates within function `setup_backoff_mock_listener` which is not recognised as a test.
    = help: Handle the `Err` variant of `std::result::Result<std::net::SocketAddr, std::net::AddrParseError>` or move the code into a test.

error: could not compile `wireframe` (lib test) due to 10 previous errors
warning: build failed, waiting for other jobs to finish...
Error: 
Error: Compilation failed with toolchain `nightly-2025-09-18`

Caused by:
    command failed: env -u CARGO -u RUSTC CLIPPY_DISABLE_DOCS_LINKS="null" DYLINT_LIBS="[\"/home/runner/.local/share/whitaker/lints/nightly-2025-09-18/x86_64-unknown-linux-gnu/lib/libwhitaker_suite@nightly-2025-09-18.so\"]" DYLINT_METADATA="null" DYLINT_NO_DEPS="0" RUSTC_WORKSPACE_WRAPPER="/home/runner/.dylint_drivers/nightly-2025-09-18/dylint-driver" RUSTUP_TOOLCHAIN="nightly-2025-09-18" "cargo" "check" "--target-dir" "/home/runner/work/wireframe/wireframe/target/dylint/target/nightly-2025-09-18" "--all-targets" "--all-features"

Stack backtrace:
   0: <anyhow::Error>::msg::<alloc::string::String>
   1: anyhow::__private::format_err
   2: <std::process::Command as dylint_internal::command::CommandExt>::success
   3: dylint::run_with_name_toolchain_map
   4: dylint::run
   5: cargo_dylint::main
   6: std::sys::backtrace::__rust_begin_short_backtrace::<fn() -> core::result::Result<(), dylint::error::ColorizedError<anyhow::Error>>, core::result::Result<(), dylint::error::ColorizedError<anyhow::Error>>>
   7: std::rt::lang_start::<core::result::Result<(), dylint::error::ColorizedError<anyhow::Error>>>::{closure#0}
   8: std::rt::lang_start_internal
   9: main
  10: <unknown>
  11: __libc_start_main
  12: _start
make: *** [Makefile:51: lint] Error 1
Error: Process completed with exit code 2.
~~~

File: crates/wireframe-verification/src/harness.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/// Build a repository-standard bounded checker for a model.
+pub fn checker<M>(model: M, bounds: VerificationBounds) -> CheckerBuilder<M>
+where
+    M: Model + Send + Sync + 'static,
+    M::State: Hash + Send + Sync,
+{
+    model
+        .checker()
+        .target_max_depth(bounds.max_depth)
+        .target_state_count(bounds.max_state_count)
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/wireframe-verification/src/harness.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 warnings, 1 inconclusive)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Title check | ⚠️ Warning | The title does not include the roadmap item reference (15.1.2) as required by the PR title check requirements when implementing a roadmapped execplan. | Update the title to include the roadmap item reference: 'Add wireframe-verification crate with placeholder Stateright model (15.1.2)' or similar. |
| Docstring Coverage | ⚠️ Warning | Docstring coverage is 76.12% which is insufficient. The required threshold is 80.00%. | Write docstrings for the functions missing them to satisfy the coverage threshold. |
| Testing (Unit And Behavioural) | ⚠️ Warning | Stateright model checking verifies properties exhaustively, but unit tests for individual state transitions, edge cases, boundary conditions, and harness variants are missing. | Add unit tests for ConnectionState helper methods. Add parametrised tests with different max\_steps bounds. Add harness edge-case tests. Add doctest examples for public API. |
| Testing (Overall) | ❓ Inconclusive | No result was produced after verification. Marking as INCONCLUSIVE. | Re-run the check or adjust instructions to produce a final result. |

File: crates/wireframe-verification/src/harness.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (leynos) [reply crates/wireframe-verification/src/harness.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: crates/wireframe-verification/src/harness.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+pub const DEFAULT_TARGET_MAX_DEPTH: NonZeroUsize = match NonZeroUsize::new(8) {
+    Some(v) => v,
+    _ => unreachable!(),
+};
+
+/// Default state budget for repository-local Stateright checks.
+pub const DEFAULT_TARGET_STATE_COUNT: NonZeroUsize = match NonZeroUsize::new(5_000) {
+    Some(v) => v,
+    _ => unreachable!(),
+};
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/wireframe-verification/src/harness.rs:17]: @coderabbitai why not `.expect(...)`?

File: crates/wireframe-verification/tests/verification_harness.rs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    },
+};
+
+#[rstest]
+fn shared_harness_runs_placeholder_model_with_explicit_bounds() {
+    assert_model_properties_with_bounds(
+        PlaceholderConnectionModel::new(6),
+        VerificationBounds {
+            max_depth: NonZeroUsize::new(8).expect("8 is non-zero"),
+            max_state_count: NonZeroUsize::new(5_000).expect("5_000 is non-zero"),
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/wireframe-verification/tests/verification_harness.rs:22]: @coderabbitai why not `.expect(...)`?

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

### ❌ Failed checks (1 error, 1 warning)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Test harness\_runs\_with\_minimal\_bounds() is vacuous: it spawns and joins a checker without calling assert\_properties(), so would pass even if properties were violated or model logic broken. | Remove vacuous test or add assert\_properties() call. Add direct tests of Model::next\_state() transitions and individual property predicates to verify core model behaviour. |
| Testing (Unit And Behavioural) | ⚠️ Warning | PlaceholderConnectionModel transition guards lack unit tests. Tests exist for ConnectionState helpers and Stateright verification, but guard-logic edge cases are untested. | Add unit tests for transitions: test apply\_enqueue None when queued, apply\_install None when non-idle/shutdown, apply\_emit None when unmet, apply\_complete None for wrong type, and verify max\_steps boundary enforcement. |
